src/monmove.cpp

```diff
@@ -271,6 +277,113 @@ float monster::rate_target( Creature &c, float best, bool smart ) const
     return INT_MAX;
 }
 
+std::map<item *, const tripoint> monster::find_loot_in_radius( const tripoint &target, int radius )
+{
+    bool steals_food = has_flag( MF_STEALS_FOOD );
+    bool steals_shiny = has_flag( MF_STEALS_SHINY );
+    bool suitable = false;
+    std::map<item *, const tripoint> lootable;
+
+    //Check all tiles nearby within radius
+    for( const tripoint &p : g->m.points_in_radius( target, radius ) ) {
+        //Check if we can see some items on the tile
+        if( g->m.sees_some_items( p, *this ) ) {
+            //A candidate tile with some items!
+            auto items = g->m.i_at( p );
+            //Check all items on tile
+            for( item &itm : items ) {
+                suitable = false;
+
+                //Does monster try to steal food and is it actually food?
+                if( steals_food && ( itm.is_comestible() && itm.get_comestible()->comesttype == "FOOD" ) ) {
+                    if( !itm.rotten() ) {
+                        suitable = true;
+                    }
+
+                    //TODO: Sanity checks here on what we're trying to loot.
+                }
+
+                //Does monster steal shiny items?
+                if( steals_shiny &&
+                    ( itm.made_of( material_id( "gold" ) ) ||
```

Comment:
Probably better to JSONize the list of stealable materials, with monster's entry having an array of materials that it'd consider stealing (if any), listing both food and non-food materials. This would also be useful, for instance, to make carnivore food stealers steal only meat-based foods they'd actually want to eat; monsters that actually eat items they stole (if they're edible at all) would have a flag noting this, so you don't have to make two mostly-separate checks for food stealing and "shinies" stealing.